cmd/fpga_plugin/dfl_test.go

```diff
@@ -188,20 +187,17 @@ func TestGetRegionDevelTreeDFL(t *testing.T) {
 
 func TestGetRegionTreeDFL(t *testing.T) {
 	expected := dpapi.NewDeviceTree()
-	hooks := []*cdispec.Hook{
-		{
-			HookName: HookName,
-			Path:     HookPath,
-		},
-	}
 	nodes := []pluginapi.DeviceSpec{
 		{
 			HostPath:      "/dev/dfl-port.0",
 			ContainerPath: "/dev/dfl-port.0",
 			Permissions:   "rw",
 		},
 	}
-	expected.AddDevice(regionMode+"-ce48969398f05f33946d560708be108a", "region1", dpapi.NewDeviceInfo(pluginapi.Healthy, nodes, nil, nil, nil, hooks))
+	cdiDevices := []*pluginapi.CDIDevice{
+		{Name: CDIKind + "=" + "region1"},
+	}
```

Comment:
Would it make sense to add hook info to the CDIDevice, so the test would look more informative?

Reply:
The hook is set in the allocate call. But the test for it is incomplete. I'll add a check to make sure the CDI device file with the hook is really written.

Reply:
Added a check to the Allocate test.

Reply:
It still looks less clear than previous implementation. Adding minimalistic cdiDevices may confuse code readers as it's not obvious that FPGA region mode requires a hook.

Reply:
I'm not sure what you mean here. Is your comment about the test or the interface between the plugin and the server(.go)?

Reply:
It's about the test. Previously hooks were explicitly passed into `AddDevice`. Now empty spec (without any hooks) is passed there.

Reply:
Maybe stating the obvious, but I think we are not on the same playing field. The flow how CDI specs are written in the case of FPGA is changing with this PR:
* FPGA plugin registers devices + cdi-device names if hooks are required.
* During Allocate, FPGA plugin checks if hooks are required and writes the cdi spec into the filesystem
  * Plugin could register the cdi-name every time and write a full cdi spec (not only the hooks)
* After Allocate, Kubelet reads the CDI device name, finds the spec and adds it/them to the container

Compared to the current (in main) flow:
* FPGA plugin registers devices and hooks
* During Allocate, server(.go) writes generic cdi spec (only hooks) to filesystem
* After Allocate, Kubelet reads the CDI device name and finds the corresponding spec etc.

Why I changed the flow:
* Only registering the name allows using pre-generated specs (initcontainer)
* DeviceTree's APi was tied to only hooks

If I change this test to register CDI spec or CDI device to DeviceTree, it will change the API of the DeviceTree. That will then affect code flows elsewhere also.

I'm not against registering full CDI spec or CDI device spec to DeviceTree, but it would force plugins to create them vs. using pre-generated ones done in initcontainer for example.